pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/Messages.properties

```diff
@@ -117,5 +117,5 @@ ModelValidatorImpl.MissingAgentParameter=Missing required parameter for agent ty
 ModelValidatorImpl.InvalidAgentParameter=Invalid config option "{0}" for agent type "{1}". Valid config options are {2}
 ModelValidatorImpl.UnknownWhenConditional=Unknown conditional {0}. Valid conditionals are: {1}
 ModelValidatorImpl.MultipleAgentParameters=Multiple parameters are required for agent type "{0}" - {1}. Please use a block instead.
-
+ModelValidatorImpl.InvalidIdentifierInEnv="{0}" is not a valid Java identifier and cannot be used for an environment variable.
```

Comment:
I would skip the mentioning of Java. Simply go with `is not a valid identifier`. No need to confuse or angry those who don't want to do the programming.